sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillPriorities.scala

```diff
@@ -58,4 +58,41 @@ object SpillPriorities {
    * Priority for multiple buffers being buffered within a call to next.
    */
   val ACTIVE_BATCHING_PRIORITY: Long = ACTIVE_ON_DECK_PRIORITY + 100
+
+  /**
+   * Priority offset for host memory buffers allocated from the pinned memory pool. They are at
+   * lower priorities, so will be spilled first, making more pinned memory available.
+   */
+  val HOST_MEMORY_BUFFER_PINNED_OFFSET: Long = -100
+
+  /**
+   * Priority offset for host memory buffers allocated from the internal pageable memory pool. They
+   * are at higher priorities than pinned memory buffers, thus making more pinned memory available;
+   * but at lower priorities than directly allocated buffers, thus freeing up the internal pageable
+   * memory pool.
+   */
+  val HOST_MEMORY_BUFFER_PAGEABLE_OFFSET: Long = 0
+
+  /**
+   * Priority offset for host memory buffers directly allocated from the OS. They are at higher
+   * priorities, thus freeing up memory pools first.
+   */
+  val HOST_MEMORY_BUFFER_DIRECT_OFFSET: Long = 100
+
+  /**
+   * Calculate the host memory buffer spill priority based on an offset, clamping it to avoid
+   * wraparound.
+   *
+   * @param originalSpillPriority the original spill priority
+   * @param offset the desired offset
+   * @return the offset spill priority, with clamping if needed
+   */
+  def getHostMemoryBufferSpillPriority(originalSpillPriority: Long, offset: Long): Long = {
+    if (offset < 0 && originalSpillPriority < Long.MinValue - offset ||
+        offset > 0 && originalSpillPriority > Long.MaxValue - offset) {
+      originalSpillPriority
```

Comment:
This should saturate to the extremes rather than ignore the offset completely when overflow would occur.
```suggestion
    if (offset < 0 && originalSpillPriority < Long.MinValue - offset) {
      Long.MinValuie
    } else if (offset > 0 && originalSpillPriority > Long.MaxValue - offset) {
      Long.MaxValue
```

Reply:
Done.